fix some issues found by enabling -Wshorten-64-to-32

hyuen opened the pull request:
Summary:
when enabling this flag, there were a lot of warnings, this pr focuses on the warnings where this comparison could be affecting array indices, which could be ones most prone to fail

the good news is that I didn't find anything obviously concerning

one degenerate case could be when the matrices we work with are too skinny could run into issues (dim1=1, dim2 needs to hold a big number)

Differential Revision: D14527182

ZolotukhinM reviewed (approved):
Overall looks good, please find some remarks inline.

hyuen commented on caffe2/operators/sparse_to_dense_mask_op.h:
```diff
@@ -43,7 +43,7 @@ class SparseToDenseMaskBase : public Operator<Context> {
   std::unordered_map<int64_t, int> sparse_;
   std::vector<int> dense_;
-  int featuresCount_;
+  size_t featuresCount_;
```
changed all the places to size_t

hyuen commented:
@pytorchbot retest

hyuen commented:
@pytorchbot retest this please

hyuen commented:
@pytorchbot retest this please

hyuen commented:
@pytorchbot retest this please

hyuen commented:
@pytorchbot retest this please

ZolotukhinM commented:
Hi Hector! Do you plan to land this or do you need any help with this?

Commit:
fix some issues found by enabling -Wshorten-64-to-32 (#18187)

Summary:
when enabling this flag, there were a lot of warnings, this pr focuses on the warnings where this comparison could be affecting array indices, which could be ones most prone to fail

the good news is that I didn't find anything obviously concerning

one degenerate case could be when the matrices we work with are too skinny could run into issues (dim1=1, dim2 needs to hold a big number)
Pull Request resolved: https://github.com/pytorch/pytorch/pull/18187

Differential Revision: D14527182

fbshipit-source-id: 42142bae9678a490267ee9055ec7c5f27e223c97
